app/main/lib/shared_models/video_model.py

```diff
@@ -14,57 +15,102 @@
 import tenacity
 import tmkpy
 from sqlalchemy.orm.exc import NoResultFound
+import boto3
+from botocore.client import Config
 
 from app.main.lib.shared_models.audio_model import AudioModel
 from app.main.lib.shared_models.shared_model import SharedModel
 from app.main.lib.similarity_helpers import get_context_query, drop_context_from_record
 from app.main.lib.helpers import context_matches
+from app.main.lib import media_crud
 from app.main.lib.error_log import ErrorLog
 from app.main import db
 from app.main.model.video import Video
+MINIO_HOST = "minio:9000"
+def download_file_from_s3(bucket: str, filename: str, local_path: str):
+    """
+    Generic download helper for s3. Could be moved over to helpers folder...
+    This function downloads a file from an S3 bucket to a local path.
+    """
+    # Set up the S3 client
+    if app.config['S3_ENDPOINT'] and MINIO_HOST in app.config['S3_ENDPOINT']:
```

Comment:
Minor: It feels wrong to have `minio:9000` hardcoded here. In particular the port should be able to change without this breaking. Can we instead look at the possibility of simply adding parameters where they are defined? I.e., if `S3_ENDPOINT` is not defined we do not set the `endpoint_url` parameter for boto3. Similar for `AWS_ACCESS_KEY_ID` etc.